### ISSUE TITLE
Collaborate with PiratenKleider

### ISSUE THREAD

**Issue by collentine** [Pirate-Parties-International/PPI-rest-api#7]
https://github.com/xwolfde/Piratenkleider

^^ wordpress theme used by several pirate parties. Implementing the API there would be very useful but guess it's mostly on their side and not on the API?

**Comment by Putr** [comment]
It's on their side to develop a plugin that display, ... what ever they wish to display. I would very much like to develop some plugins, but I'm in dire need of skilled front-end (CSS, JS is not that big of a deail) developers. PPSI has a number of them (one is helping now) but most of them are very green (that's why they are working for free) and already very busy with our own projects (like nadzorovan.si which will be given to all other parties to deploy as their own).

I'm very much willing to help implement anything in PHP or extend the API to include endpoints that any plugin developer mite need, but some other party will have to supply the frontend developer for those, out of scope, ideas.

**Comment by PeterTheOne** [comment]
@collentine: Could you write in more detail what a wordpress plugin/theme should display or how it could be used? Maybe I can find some time to make something happen.

**Comment by collentine** [comment]
@PeterTheOne mostly just making the API calls easily accessible to people not used to working with API's. 

e.g. dropdown lists or a way of adding several "key" stats of a country in an easy way. One example would be someone writing about a pirate party and creating a "country box" that could be added to bottom of articles with some stats. Another would be just developing a side-bar widget to fetch data from API and display all Pirate Parties (maybe with option of choosing what info to display - soc media, websites etc)

Widget example = something similar to what is displayed on http://pp-international.net/ but fetching stats from API to make sure they are up to date.

**Comment by PeterTheOne** [comment]
@collentine I've startet something like this just now, check it out at https://github.com/PeterTheOne/wp-pirate-parties

**Comment by Putr** [comment]
@PeterTheOne how far you with that? Do you need anything for the wp integration from the API side?

**Comment by PeterTheOne** [comment]
For the older version of the data it works fine, haven't tested it since. But I will fix what is broken. It does some filtering and sorting that could be done by the API, otherwise it works just fine.

filters:
get parties who are (full, observer or not) member of ppi, ppeu or both.

sort:
by "native" party name
by english party name
by country name

**Comment by Putr** [comment]
Yes sorting and filtering will be an easy addition (and a core one at that) when I migrate the data into a proper DB. At the moment it's using redis which is a tad harder to work with. I used it because I did not thing we would be doing anything but displaying data at the time.

**Comment by PeterTheOne** [comment]
Other filters we may want to add:
filter by regional or national parties
filter by continent
filter by country (for regional parties)

**Comment by PeterTheOne** [comment]
@collentine I've updated the wordpress plugin, it works with the newest API version. To install just copy the files to ``*wordpress-install-dir*/wp-content/plugins/wp-pirate-parties``, activate the plugin and add the widget to the sidebar.
